statsmodels/genmod/generalized_linear_model.py

```diff
@@ -605,6 +604,8 @@ def hessian(self, params, scale=None, observed=True):
         hessian : ndarray
             Hessian, i.e. observed information, or expected information matrix.
         """
+        if hasattr(self, '_optim_hessian') and self._optim_hessian == 'eim':
```

Comment:
switch to `observed=None` in signature.
and protect this with `if observed is None:   ....  else observed=True`